src/compiler/compiler.rs

```diff
@@ -1008,11 +1012,30 @@ diab
         for line in stdout.lines() {
             //TODO: do something smarter here.
             match line {
```

Comment:
We should be able to reduce duplication here by:

```
match line {
  "clang" | "clang++" => {
    ... Clang { clangplusplus: line == "clang++" }
  // and similarly for the gcc/g++ case.
```

it does duplicate computation, but I am not super concerned about the hotness of this code.